src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,7 +32,18 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        //this.value = trimmedAddress;
+
+        String[] addressArray = trimmedAddress.split(DELIMITER);
+        this.block = new Block(addressArray[0]);
+        this.street = new Street(addressArray[1]);
+        this.unit = new Unit(addressArray[2]);
+        this.postalCode = new PostalCode(addressArray[3]);
+
+        this.value = block.getBlockNo() + DELIMITER +
```

Comment:
If a value can be easily derived from other values, consider not storing it to minimise redundant data.